### DISCUSSION TITLE
Set an encrypted password instead of plain text via the command line

### DISCUSSION THREAD

**Discussion by ThomasX-1970** [rustdesk/rustdesk#9229] [General]
Hello,
we distribute RustDesk (community version) with our software distribution on Windows PCs that control/monitor machines. A session should be possible if the password is known, nobody should have to confirm control by clicking. We want to use a permanent password for this.
I have found two ways to make this possible:

1. With the command **rustdesk.exe --password PW123456!**
2. Enter the password in plain text in the toml file. This is automatically encrypted after RustDesk is started

Unfortunately, the **passwords are stored in plain text** in the directories of the software distribution (on the server and each client) and can therefore be read by any user.

I tried to transfer the encrypted password and salt from a test installation (toml file) to another installation, but unfortunately that doesn't work. Presumably a unique identifier (ID, hardware hash ...) of the local machine is used to encrypt the password?

Would it be possible to add a command to Rustdesk to set an encrypted password on the client, e.g.: **rustdesk.exe --password-enc “my encrypted password”**?

Many thanks and best regards
Thomas

**Comment by basilgello** [comment]
Indeed transferring TOML files will work only on Android and iOS: https://github.com/rustdesk/rustdesk/discussions/9237

For the password security, it is better to implement password from stdin or environment variable like in OpenSSL.
